datafusion/common/src/error.rs

```diff
@@ -417,6 +414,11 @@ impl DataFusionError {
         // return last checkpoint (which may be the original error)
         last_datafusion_error
     }
+
+    /// wraps self in Self::Context with a description
+    pub fn context(self, description: impl Into<String>) -> Self {
```

Comment:
Yay, thank you! I'm happy to see this being used and improved.

Reply:
Yeah, it is really nice -- thanks for adding it!